packages/flutter/lib/src/foundation/memory_allocations.dart

```diff
@@ -297,6 +315,26 @@ class FlutterMemoryAllocations {
     dispatchObjectEvent(ObjectDisposed(object: object));
   }
 
+  /// Create an object with exemption from disposal.
+  ///
+  /// All objects created by [builder] will be exempt from disposal.
+  ///
+  /// The method is useful for creating singletons:
+  ///
+  /// ```dart
+  /// final ValueNotifier<bool> mySingleton = FlutterMemoryAllocations.instance.exemptFromDisposal(() {
+  ///   return ValueNotifier<bool>(false);
+  /// });
+  /// ```
+  T exemptFromDisposal<T>(ObjectBuilderCallback<T> builder ){
+    ObjectCreated._exemptionFromDisposal ++;
+    try {
+      return builder();
+    } finally {
+      ObjectCreated._exemptionFromDisposal --;
+    }
+  }
+
```

Comment:
Does this work in the generic case? What if as a side effect of creating my long-lived singleton object some short-lived disposables are created (e.g. in the constructor of the singleton)? Wouldn't those also now be incorrectly exempt from disposal?

Reply:
Theoretically it may be that there will be false negatives. Practically I cannot come up with realistic examples.
And, anyway, leak_tracker, as well as any other testing, cannot guarantee all bugs being caught.
Good news is that it is not often case. I found just two cases across FF.
This implementation seems to be the best balance between simplicity and reliability.

Other alternatives I considered:

1. Add exemptions to the leak tracking configuration, instead of the actual code that creates the objects. 
Cons are:
a. Creator of the disposable is one to take care about disposing or explicit declaration that it is not needed.
b. It is non-trivial to specify in settings which object to exempt

2. Propagate the exemption flag down the tree to all disposable members that need to be exempted. Seems to be unnecessary complexity, because all members are normally not disposed.

How does it sound?

Reply:
> leak_tracker, as well as any other testing, cannot guarantee all bugs being caught.

We still should avoid adding APIs that have a built-in footgun that will make it easier to implement a bug.

Reply:
True. Do you have in mind an alternative?

Reply:
I think this is a potentially concerning footgun. 
For example, 
```FlutterMemoryAllocations.instance.exemptFromDisposal(() => ValueNotifier<bool>(WidgetBindings.instance.someField));```
might accidentally mark a bunch of disposables created as part of instantiating the WidgetBindings.instance singleton as disposable.